packages/http-client-java/generator/http-client-generator-core/src/main/java/com/microsoft/typespec/http/client/generator/core/mapper/PagingMetadata.java

```diff
@@ -138,15 +147,53 @@ private static ModelPropertySegment getPageableNextLink(XmsPageable xmsPageable,
         return ClientModelUtil.getModelPropertySegment(responseBodyType, xmsPageable.getNextLinkName());
     }
 
-    static MethodPageDetails.ContinuationToken getContinuationToken(XmsPageable xmsPageable, IType responseBodyType) {
-        // TODO: anu change this method to 'private' once ClientCoreClientMethodMapper is refactored.
+    /**
+     * Inspects the given {@link ClientMethodParametersDetails} for a code model query {@link Parameter} representing
+     * the page size (commonly named "maxpagesize" or "maxPageSize"), and returns the corresponding
+     * {@link ClientMethodParameter}.
+     *
+     * @param parametersDetails the {@link ClientMethodParametersDetails} to inspect.
+     * @return the {@link ClientMethodParameter} representing the page size, or {@code null} if not found.
+     */
+    private static ClientMethodParameter
+        getPageSizeClientMethodParameter(ClientMethodParametersDetails parametersDetails) {
+        return parametersDetails.getParameterTuples().filter(t -> {
+            final Parameter cmParameter = t.codeModelParameter;
+            final Protocols protocol = cmParameter.getProtocol();
+            final boolean isQuery = protocol != null
+                && protocol.getHttp() != null
+                && protocol.getHttp().getIn() == RequestParameterLocation.QUERY;
+            if (!isQuery) {
+                return false;
+            }
+            final String serializedName = cmParameter.getLanguage().getDefault().getSerializedName();
+            if (Objects.equals(serializedName, "maxpagesize")) {
+                return true;
+            }
+            //
+            // Note: The below fallback logic using Java name exists because, according to the REST API Guidelines for
+            // Swagger, the standard query parameter name for page size is "maxpagesize".
+            // However, some services were designed before this guideline and may use a different serialized name (e.g.,
+            // "$maxpagesize" or "maxPageSize").
+            // In such cases, spec authors were instructed to add directive to make it "maxpagesize" that produces
+            // "maxPageSize" Java name in the SDK.
+            //
```

Comment:
Hi @weidongxu-microsoft, I am documenting the context you provided offline. Does this look correct? Between do we know the exact directive used or this?

Reply:
found [this](https://github.com/Azure/azure-rest-api-specs/blob/f677519d3a8beabfb0a42b738d33e0d4d707c3d1/specification/batch/Azure.Batch/common.tsp#L355) one, where Java name is defined to be "maxPageSize"

Reply:
Ah, yes, it's them, Batch.